### ISSUE TITLE
Proposal `cypress init` command

### ISSUE THREAD

**Issue by brian-mann** [cypress-io/cypress#619]
We no longer seed anything when running `cypress run` on a fresh project. 

Instead we now error: #618 

For that reason we should give users the ability to seed / scaffold out files from the command line instead of using the GUI.

Proposal add a `cypress init` command which does this.

Bonus points if we offer an `inquirer` view enabling the user to select what they want to. We could even explain what each one does.

**Comment by Knaledge** [comment]
@brian-mann - I am running into this exact (*exact*) situation right now. I am so glad I found this specific issue entry here! It captures exactly what I'm running into. 

Is there any ETA? 

We do not have the desire to gen node_modules per project (at the moment, or even short term) as we will only be utilizing a single version of Node.js and Cypress for the foreseeable future. 

Or maybe that logic is misapplied. Either way - we went from a "global" installation to a "local" installlation (Cypress CLI was complaining that the .cache folder did not contain a "Cypress" instance) and it worked great. 

We now do this: 

`su - cypress -c 'npm install cypress'`

I then do the following, and encounter the "cypress.json" error mentioned here in this issue: 

`[cypress@proto ~]$ npx cypress run --spec '/home/cypress/cy.tests/EXAMPLE.staff_login.js'
It looks like this is your first time using Cypress: 3.0.2

 ✔  Verified Cypress! /home/cypress/.cache/Cypress/3.0.2/Cypress

Opening Cypress...
Could not find any tests to run.

We looked but did not find a cypress.json file in this folder: /home/cypress` 

Passing --config (despite no test/spec specified) further underscores this: 

`[cypress@proto ~]$ npx cypress run --config integrationFolder=/home/cypress/cy.tests/,fixturesFolder=false
Could not find any tests to run.

We looked but did not find a cypress.json file in this folder: /home/cypress`

**Comment by Knaledge** [comment]
@brian-mann - Alternatively, just some clear instruction/guidance on how to do this "init" on my own would be so, _so_ helpful as I believe it is the last barricade. Cypress docs - while super awesome (really!) - don't seem to cover what exactly is required for a run to work. It seems there is a lot of reliance on using the UI to select a project (which then generates the required files/folders/etc.) - but as you raised here, we're using CLI entirely. 

Even just a few quick pointers to get me on my feet would be incredibly appreciated.

**Comment by jennifer-shehane** [comment]
Looping the below suggestion in as discussed in https://github.com/cypress-io/cypress/issues/164:

>Never seed tests when running `cypress ci`

**Comment by jennifer-shehane** [comment]
- Allow user to choose a different directory other than `cypress` directory to scaffold?? (we could set up the appropriate mappings in their `cypress.json`
- Offer prompt to skip interactive mode just doing `cypress init -f` like `npm init`.
- Add new command to Command Line docs.
- `cypress init` should be released with GUI option, but can be developed after we figure out user flow of GUI: https://github.com/cypress-io/cypress/issues/3364
- Make the `configFile` optional -- (kind of required for the Dashboard.....), so only make it if they set up Dashboard.
- Make the `pluginsFile` and `supportFile` optional also.
- Choose whether they want to seed or not & reinvoke it as part of GUI later.
- Relax rules of scaffolding/seeding - don't regenerate `cypress` folder. 
- Don't error on `cypress run` if no tests - improve this message and give a call to action.

**Comment by ollie-o** [comment]
This issue was opened in 2017. What is the ETA? Is someone working on it?

Also, are there any workarounds in the meantime?

**Comment by Bkucera** [comment]
another use case would be properly setting up `eslint` with `eslint-plugin-cypress`, since there can be quite a few steps to this e.g.:

- create `.eslintrc` in cypress directory
- if users are using typescript, they will need @ typescript-eslint/parser + plugin

**Comment by bahmutov** [comment]
@Bkucera we could just implement / add this to https://github.com/bahmutov/cly issues - there the progress will be much faster

PS: @ollie-o

**Comment by Adil-Iqbal** [comment]
I've had to train a few devs on the team to use Cypress, and not having this feature kind of threw them for a loop. It's just more intuitive to explain.

"First open the test launcher. Then close the test launcher. Now we can create a test..."

vs.

"Run the intialize command. Now we can create a test..."

All I'm saying is, having this command would really help with our onboarding process.

**Comment by benjaminbours** [comment]
@Adil-Iqbal president!

It is also very nice for the motivation explained here #1929

**Comment by the0neWhoKnocks** [comment]
It's a shame this issue hasn't gotten the attention it deserves. In case anyone else wanders across this ticket, and needs a zero-dependency solution:
```sh
#!/bin/bash

TEST_FOLDER="./e2e"
PATH_01=("${TEST_FOLDER}/cypress.json" "{ \"video\": false }\n")
PATH_02=("${TEST_FOLDER}/cypress/integration")
PATH_03=("${TEST_FOLDER}/cypress/integration/example.test.js" "context('Example', () => {\n  beforeEach(() => { cy.visit('/'); });\n\n  it('should have loaded', () => {\n    cy.get('title').contains(/.*/);\n  });\n});\n")
scaffold=(PATH_01[@] PATH_02[@] PATH_03[@])
if [ ! -f "${!scaffold[0]:0:1}" ]; then
  echo "[SCAFFOLD] Cypress test directory"
  
  length=${#scaffold[@]}
  for ((i=0; i<$length; i++)); do
    path="${!scaffold[i]:0:1}"
    contents="${!scaffold[i]:1:1}"
    
    if [[ "${contents}" != "" ]]; then printf "${contents}" > "${path}"; else mkdir -p "${path}"; fi
  done
fi
```

**TLDR**: If the `cypress.json` doesn't exist in the `TEST_FOLDER`, scaffold the bare minimum of what's required to get tests running.

**Comment by MikeMcC399** [comment]
- Added from duplicate #29619 

### What would you like?

Cypress should support an automated way of configuring and setting up E2E examples through the command line, for instance using an environment variable.

The choice of the full set of "Scaffold example specs" or the single test spec from "Create new spec" should be given.

### Why is this needed?

Currently it is necessary to use `cypress open`, then to manually agree to the configuration, followed by selecting spec options.

This means that there is no straightforward way to automate the setup or update of a running Cypress example project. Instead, manual steps have to be described and carried out. An example of this is in the instructions [factory/test-project/README.MD](https://github.com/cypress-io/cypress-docker-images/blob/master/factory/test-project/README.MD) in the [cypress-io/cypress-docker-images](https://github.com/cypress-io/cypress-docker-images) repo, where it is necessary to combine shell instructions with manual instructions to achieve setup:

---
```shell
rm -rf cypress cypress.config.js
npm install cypress@latest --no-package-lock
npx cypress open
```

- Select "E2E Testing"
- Select "Electron" browser
- Select "Continue"
- Select "Scaffold example specs"
- Close all Cypress windows

Test that scaffolded specs run:

```shell
npm test
```
---

**Comment by MikeMcC399** [comment]
- This issue becomes more important if https://github.com/cypress-io/cypress/issues/33560 goes ahead.
